addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/filesystem-role-service.js

```diff
@@ -0,0 +1,553 @@
+    await super.init();
+    const [dbService] = await this.service(['dbService']);
+    const table = this.settings.get(settingKeys.tableName);
+
+    this._getter = () => dbService.helper.getter().table(table);
+    this._updater = () => dbService.helper.updater().table(table);
+    this._query = () => dbService.helper.query().table(table);
+    this._deleter = () => dbService.helper.deleter().table(table);
+    this._scanner = () => dbService.helper.scanner().table(table);
+  }
+
+  /**
+   * This method returns the filesystem role entity.
+   *
+   * @param requestContext The standard requestContext
+   * @param accountId The data source account id that the filesystem role belongs to
+   * @param bucket The name of the bucket
+   * @param arn The filesystem role arn
+   * @param fields An array of the attribute names to return, default to all the attributes of the filesystem entity
+   */
+  async find(requestContext, { accountId, bucket, arn, fields = [] }) {
+    // Perform default condition checks to make sure the user is active
+    await this.assertAuthorized(
+      requestContext,
+      { action: 'read', conditions: [allowIfActive, allowIfAdmin] },
+      { accountId, arn },
+    );
+
+    const dbEntity = await this._getter()
+      .key(fsRoleIdCompositeKey.encode({ accountId, bucket, arn }))
+      .projection(fields)
+      .get();
+
+    return toFsRoleEntity(dbEntity);
+  }
+
+  /**
+   * This method is similar to the 'find' method but it throws an exception if the filesystem role is not found.
+   *
+   * @param requestContext The standard requestContext
+   * @param accountId The data source account id that the filesystem role belongs to
+   * @param bucket The name of the bucket
+   * @param arn The filesystem role arn
+   * @param fields An array of the attribute names to return, default to all the attributes of the
+   * filesystem role entity.
+   */
+  async mustFind(requestContext, { accountId, bucket, arn, fields = [] }) {
+    const result = await this.find(requestContext, { accountId, bucket, arn, fields });
+    if (!result) throw this.boom.notFound(`Filesystem role with arn "${arn}" does not exist`, true);
+    return result;
+  }
+
+  /**
+   * Call this method to allocate a filesystem role entity for the given study. This method is smart
+   * enough to reuse an existing filesystem role entity if there is one already for the same study.
+   *
+   * Note: this method creates an IAM role resource in an AWS account, if needed.
+   *
+   * @param requestContext The standard requestContext
+   * @param studyEntity The study entity. IMPORTANT, it is expected that this study entity will have
+   * an additional property named 'envPermission', it is an object with the following shape:
+   * { read: true/false, write: true/false }
+   * @param environmentEntity The environment entity that will use the filesystem role
+   */
+  async allocateRole(requestContext, studyEntity = {}, environmentEntity = {}, memberAccountId) {
```

Comment:
Nit: Optional params should come after required params

Reply:
Let me add memberAccountId = '',   studyeEntity, envrionmentEntity, memberAccountId  all are not optional.  I do the studyEntity = {} trick to help avoid null pointer exceptions while I check the values.  In my @param I don't specify that these are optional.  So, once we produce jsdoc out of the code, you will see that all my jdoc have @param and if they are optional, I will "hopefully" remember to mention that :)